### PR TITLE
Support PR workflow close handler

### PR DESCRIPTION
**Pull request by samrabelachew**
Implements the gateway logic to listen for close events and forwards them to the Temporal worker to process.

### CODE REVIEW

File: server/neptune/gateway/event/closed_pull_request_handler.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		c.Logger.ErrorContext(ctx, "unable to allocate pr mode")
+		return nil
+	}
+	if !shouldAllocate {
+		c.Logger.InfoContext(ctx, "handler not configured for allocation")
+		return nil
+	}
+	return c.PRCloseSignaler.SignalWorkflow(
+		ctx,
+		pr.BuildPRWorkflowID(event.Pull.BaseRepo.FullName, event.Pull.Num),
```
Comment (nishkrishnan) [inline server/neptune/gateway/event/closed_pull_request_handler.go:49]: isn't this going to have the same problem we just talked about? ie. we receive closed events for all PRs and we need to know if a workflow exists for a given PR

Comment (samrabelachew) [reply server/neptune/gateway/event/closed_pull_request_handler.go:49]: Yep, exactly. Forgot to add a comment here, but I was going to update it once we determine best way to check if a workflow exists

Comment (samrabelachew) [reply server/neptune/gateway/event/closed_pull_request_handler.go:50]: filtering out serviceerror.NotFound

Comment (nishkrishnan) [reply server/neptune/gateway/event/closed_pull_request_handler.go:50]: we should add metrics around how often we're seeing this.

Comment (nishkrishnan) [reply server/neptune/gateway/event/closed_pull_request_handler.go:50]: Also worth thinking about how we will know whether service not found is something to ignore vs something we need to take action on.

Comment (samrabelachew) [reply server/neptune/gateway/event/closed_pull_request_handler.go]: Added a metric, but I don't think we need to do much outside of of observability. Temporal confirms a call to a non-running workflow should return this error type, so we should ignore the underlying Github event. If there was ever a case we return this for the wrong reasons (a workflow does exist and is running), the workflow itself has a close event poller + full workflow timeout to shut itself down.

**Review by nishkrishnan — APPROVED** [review]

**Pull request merged**